src/Select/Select.tsx

```diff
@@ -1,107 +1,52 @@
-import { styles } from './shared';
-export type {
-  ValueType,
-  ActionMeta,
-  GroupTypeBase,
-  OptionTypeBase
-} from 'react-select';
-
-export type SelectOptionType<OptionType extends OptionTypeBase> =
-  | OptionType
-  | OptionsType<OptionType>;
-
-export interface SelectProps<
-  OptionType extends OptionTypeBase,
-  IsMulti extends boolean,
-  GroupType extends GroupTypeBase<OptionType>
-> extends SelectBaseProps<OptionType, IsMulti, GroupType> {
-  icon?: boolean;
-  clearText?: string;
+import React, { ChangeEvent, ReactElement } from 'react';
+import { Input } from 'reactstrap';
+
+export interface SelectProps {
+  /**
+   * L'identificativo univoco del componente
+   */
+  id: string;
+  /**
+   * Se `true` determina la disabilitazione del campo
+   */
+  disabled?: boolean;
+  /**
+   * L'etichetta del campo
+   */
+  label: string;
+  children: ReactElement<'option'>[];
+  /**
+   * Gestisce il cambio del valore selezionato
+   * @param selectedValue
+   * @returns
+   */
+  handleChange: (selectedValue: string) => void;
 }
 
-const DropdownIndicator = <
-  OptionType extends OptionTypeBase,
-  IsMulti extends boolean,
-  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
->(
-  props: IndicatorProps<OptionType, IsMulti, GroupType>
-) => {
-  return (
-    <components.DropdownIndicator {...props}>
-      <Icon icon='it-arrow-down-triangle' aria-hidden />
-    </components.DropdownIndicator>
-  );
-};
-
-const Option = <
-  OptionType extends OptionTypeBase,
-  IsMulti extends boolean,
-  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
->(
-  props: OptionProps<OptionType, IsMulti, GroupType>
-) => {
-  return (
-    <div className='select-pill text-primary'>
-      <components.Option {...props} />
-    </div>
-  );
-};
-
-const CustomClearText: FC<{ clearText: string }> = ({ clearText }) => (
-  <>{clearText}</>
-);
-
-export const Select = <
-  OptionType extends OptionTypeBase = { label: string; value: string },
-  IsMulti extends boolean = false,
-  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
->({
-  clearText = 'Annulla',
-  icon = false,
-  ...props
-}: SelectProps<OptionType, IsMulti, GroupType>): ReactElement<
-  SelectProps<OptionType, IsMulti, GroupType>
-> => {
-  // @TODO: move it outside when migrating to hooks
-  const ClearIndicator = <
-    OptionType extends OptionTypeBase,
-    IsMulti extends boolean,
-    GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
-  >(
-    clearIndicatorProps: IndicatorProps<OptionType, IsMulti, GroupType>
-  ) => {
-    const {
-      children = <CustomClearText clearText={clearText} />,
-      innerProps: { ref, ...restInnerProps }
-    } = clearIndicatorProps;
-    return (
-      <div className='select-pill text-primary' {...restInnerProps} ref={ref}>
-        <div style={{ padding: '0px 5px' }}>{children}</div>
-      </div>
-    );
+export const Select = ({
+  id = 'select',
```

Comment:
I think it is not a good idea to have a default id value, as `id` is something that should be unique in a DOM tree.
If the only reason to have it is to use it as reference for the `label` tag, I could suggest branching the rendering based on the `id` value:
* if none is passed, then the `label` tag will wrap the `Input` component
* otherwise use the current structure (without the `-select` suffix)

Reply:
I agree with you, my bad for not knowing `label` could wrap its component